python-threatexchange/threatexchange/fetcher/fetch_api.py

```diff
@@ -199,3 +204,56 @@ def report_false_positive(
                 tags=set(),
             ),
         )
+
+
+class SignalExchangeAPIWithIterFetch(
+    t.Generic[
+        state.TFetchCheckpoint
+    ],  # TODO move to SignalExchangeAPI in a future diff
+    SignalExchangeAPI,
+):
+    """
+    An alternative fetch_once implementation simplifies tracking state
+    """
+
+    def __init__(self) -> None:
+        self._fetch_iters: t.Dict[
+            str, t.Iterator[state.FetchDelta[state.TFetchCheckpoint]]
+        ] = {}
+
+    def fetch_once(  # type: ignore[override]  # fix with generics on base
+        self,
+        supported_signal_types: t.List[t.Type[SignalType]],
+        collab: CollaborationConfigBase,
+        # None if fetching for the first time,
+        # otherwise the previous FetchDelta returned
+        checkpoint: t.Optional[state.TFetchCheckpoint],
+    ) -> state.FetchDelta[state.TFetchCheckpoint]:
+        it = self._fetch_iters.get(collab.name)
+        if it is None:
+            it = self.fetch_iter(supported_signal_types, collab, checkpoint)
+            self._fetch_iters[collab.name] = it
+        delta = next(it, None)
+        # This can happen if the last yielded element did not set done
+        # which will cause next() to be called again after the iterator
+        # is exhaused
+        assert delta is not None, "fetch_iter stopping yielding too early"
+        return delta
+
+    def fetch_iter(
+        self,
+        supported_signal_types: t.List[t.Type[SignalType]],
+        collab: CollaborationConfigBase,
+        # None if fetching for the first time,
+        # otherwise the previous FetchDelta returned
+        checkpoint: t.Optional[state.TFetchCheckpoint],
+    ) -> t.Iterator[state.FetchDelta[state.TFetchCheckpoint]]:
+        """
+        An alternative fetch_once implementation simplifies tracking state.
```

Comment:
stale docstring?

Reply:
This appears to be the right docstring - can you elaborate?